doc/user/content/overview/key-concepts.md

```diff
@@ -10,45 +10,88 @@ aliases:
   - /overview/api-components/
 ---
 
-Materialize is a streaming database with a SQL API. However, despite the fact
-that Materialize uses SQL idioms and can process data from databases, it
-actually has very little in common with "databases" as most people think of them.
+Most physical and cloud-hosted database solutions require you to configure
```

Comment:
I think we need to more directly introduce the purpose of this page, and not focus on the benefits of a managed service in general. Suggestion (to start from):

```suggestion
This page introduces the key objects of Materialize.  
```

We might want to also say something like:

```
Some of these objects correspond traditional SQL idioms but have different properties and behaviors in the context of a streaming database like Materialize.
```